include/oneapi/dpl/internal/distributed_ranges_impl/detail/ranges.hpp

```diff
@@ -359,17 +346,19 @@ struct local_fn_
     }
 
     template <typename T>
-    decltype(auto)
+    auto
     operator()(T&& t) const
     {
-        return t;
+        return std::forward<T>(t);
     }
 };
 
-inline constexpr auto local = local_fn_{};
-
 } // namespace __detail
 
+inline constexpr auto segments = __detail::segments_fn_{};
+
+inline constexpr auto local_or_identity = __detail::local_or_identity_fn_{};
+
```

Comment:
`local_or_identity` is seemingly a helper for internal use; should it really be added into the same namespace with the public APIs? I would keep it in `__detail`.

Reply:
The same for
```
inline constexpr auto local = __detail::local_fn_{};
inline constexpr auto segments = __detail::segments_fn_{};
```
?
Do you think we should also put them in detail? I think not. Regarding _segments_ I it is a public interface. Local is public interface too. 

And regarding local_or_identity I think it looks better if it is in the same namespace as local and segments. But if you have strong feeling it should be in __detail I can move (leaving segments and local outside __detail). Please decide.

Reply:
Maybe `local_or_identity` coud even be moved into another file, if there is a suitable place. That would make the distinction even clearer.

Reply:
moved to __detail